### PR TITLE
Converts float to double

### LINKED ISSUE #2322: Finish "convert floats to double" PR
Here is a weird one. PR was approved then not merged. It now has conflicts and no activity for over a year. It might just need some minor updates: https://github.com/ros-planning/moveit2/pull/1578

### PR DESCRIPTION
**Pull request by Shobuj-Paul**
### Description

Fixed merge conflicts by keeping the previous PR changes where float was converted to double, and accepted current changes for the rest.

Fixes #2322 

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit2/blob/main/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://moveit.picknik.ai/humble/doc/examples/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by Shobuj-Paul** [comment]
Few changes are required for this PR, currently making those.

**Comment by Shobuj-Paul** [comment]
Changes done. I am not very clear as to what this PR was originally supposed to accomplish, but this fix builds!

**Comment by ibrahiminfinite** [comment]
There are some clang-tidy complaints, could you fix those ?

**Comment by Shobuj-Paul** [comment]
> There are some clang-tidy complaints, could you fix those ?

Yep, done. Just needed to change a couple of variable names in GLRenderer.

File: moveit_core/collision_detection/src/collision_octomap_filter.cpp [hunk 1]
```diff
@@ -211,13 +211,12 @@ bool findSurface(const octomap::point3d_list& cloud, const double& spacing, cons
...
 {
-  const double epsilon = 1e-10;
   const int iterations = 10;
-  double intensity = 0;
-
   octomath::Vector3 p = seed, dp, gs;
   for (int i = 0; i < iterations; ++i)
   {
+    const double epsilon = 1e-10;
+    double intensity = 0;
```
[the remark this answers is not in the document]
Comment (Shobuj-Paul) [reply moveit_core/collision_detection/src/collision_octomap_filter.cpp]: Fixed.

[the remark this answers is not in the document]
Comment (Shobuj-Paul) [reply moveit_core/collision_detection/src/collision_octomap_filter.cpp]: Fixed.

**Review by tylerjw — APPROVED** [review]

**Pull request merged**